src/components/Card.js

```diff
@@ -42,8 +42,8 @@ class Card {
     if (this._owner._id === this._currentUserId) {
       deleteButton.style.display = "block"; // Show the trash button
       deleteButton.addEventListener("click", () => {
-        this._handleDelete(this, this._cardID);
-        deleteButton.remove();
+        this._deleteCard(this, this._cardID); // Perform the delete action
+        deleteButton.parentElement.removeChild(deleteButton); // Remove the delete button
```

Comment:
Needs correction: Please remove this line, you need to remove the trash button on line 49 using `deleteButton.remove()`, not here